### PR TITLE
format logging from port-forward error

### LINKED ISSUE #117076: Incorrect logging from port-forward error
### What happened?

I got an error like

```
 error forwarding port 15008 to pod ef373fd9bb0eace525a5232b56196e8fd7144228d6c533fb645d206a134e8245, uid : network namespace for sandbox "ef373fd9bb0eace525a5232b56196e8fd7144228d6c533fb645d206a134e8245" is closed
```

### What did you expect to happen?

My error is formated like 

```
 error forwarding port 15008 to pod SOME_POD_NAME, uid SOME_UID: network namespace for sandbox "ef373fd9bb0eace525a5232b56196e8fd7144228d6c533fb645d206a134e8245" is closed
```

However, at the callsite we don't set the UID and we set "Pod Name" to the sandbox ID: https://github.com/kubernetes/kubernetes/blame/d89d5ab2680bc74fe4487ad71e514f4e0812d9ce/pkg/kubelet/cri/streaming/server.go#L351-L361

### How can we reproduce it (as minimally and precisely as possible)?

Have a failed port-forward

### Anything else we need to know?

_No response_

### Kubernetes version

<details>

```console
$ kubectl version
# paste output here
```

</details>

### Cloud provider

<details>

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

### PR DESCRIPTION
**Pull request by cyclinder**
#### What type of PR is this?

/kind bug
#### What this PR does / why we need it:

format logging from port-forward error:

```
 error forwarding port 15008 to pod ef373fd9bb0eace525a5232b56196e8fd7144228d6c533fb645d206a134e8245, uid : network namespace for sandbox "ef373fd9bb0eace525a5232b56196e8fd7144228d6c533fb645d206a134e8245" is closed
```

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubernetes/issues/117076

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by bart0sh** [comment]
@cyclinder please fix CI test failures, thanks.

**Comment by cyclinder** [comment]
@bart0sh thanks for the reminder, I'm trying to fix CI test failures and see what went wrong.

File: staging/src/k8s.io/cri-api/pkg/apis/runtime/v1/api.proto [hunk 1]
```diff
@@ -1360,8 +1360,14 @@ message AttachResponse {
...
     // ID of the container to which to forward the port.
     string pod_sandbox_id = 1;
+    // Name of pod
+    string pod_name = 2;
+    // Namespace of pod
+    string pod_namespace = 3;
+    // Uid of pod
+    string pod_uid = 4;
     // Port to forward.
-    repeated int32 port = 2;
```
Comment (tzneal) [inline staging/src/k8s.io/cri-api/pkg/apis/runtime/v1/api.proto:1364]: I don't think you can do this, changing the proto tags breaks compatibility.

Comment (cyclinder) [reply staging/src/k8s.io/cri-api/pkg/apis/runtime/v1/api.proto:1364]: Thanks for the review, CI test failures may be caused by this. Now I'll update it and see if the CI works.

**Comment by aojea** [comment]
are we going to modify the CRI API just for this?

**Comment by cyclinder** [comment]
I don't think it's necessary, but maybe need CRI or sig/node maintainers ack? 

/cc @bart0sh @SergeyKanzhelev plz

**Comment by aojea** [comment]
I suggest to not modify the API

**Comment by cyclinder** [comment]
Can you tell me why? 🤔  If we need to show basic information about the pod in the port-forward error message, then we need to add fields to the API. Otherwise, there is no better way to get the basic information of the pod.

**Comment by tzneal** [comment]
> Can you tell me why? 🤔 If we need to show basic information about the pod in the port-forward error message, then we need to add fields to the API. Otherwise, there is no better way to get the basic information of the pod

IMO, changing the CRI is rather disruptive requiring coordinating changes with container runtimes.  Doing so in order to improve an error message doesn't seem worth it.

**Comment by cyclinder** [comment]
Thanks for the explanation @tzneal, If we can't change the API, improving an error message is harder...I tried other ways to see if it would work, and If someone can give me advice it would be appreciated.

**Comment by marquiz** [comment]
> IMO, changing the CRI is rather disruptive requiring coordinating changes with container runtimes. Doing so in order to improve an error message doesn't seem worth it.

Yeah, changing the CRI API just for this doesn't feel right.

Isn't there any reasonable way to get that information on the kubelet side directly?

**Pull request closed without merging**